packages/services/api/src/modules/shared/module.graphql.ts

```diff
@@ -7,12 +7,22 @@ export default gql`
   scalar JSONSchemaObject
   scalar SafeInt
 
+  extend schema
+    @link(url: "https://specs.apollo.dev/link/v1.0")
+    @link(url: "https://specs.apollo.dev/federation/v2.0", import: ["@tag"])
+
+  directive @link(url: String!, import: [String!]) repeatable on SCHEMA
+
+  directive @tag(
+    name: String!
+  ) repeatable on FIELD_DEFINITION | INTERFACE | OBJECT | UNION | ARGUMENT_DEFINITION | SCALAR | ENUM | ENUM_VALUE | INPUT_OBJECT | INPUT_FIELD_DEFINITION | SCHEMA
+
   type Query {
-    noop: Boolean
+    _: Boolean @tag(name: "public")
   }
 
   type Mutation {
-    noop(noop: String): Boolean
+    _: Boolean @tag(name: "public")
```

Comment:
I'm not very familiar with the contract logic.
if the schema is not "public", then is this mutation empty or does the entire Mutation type get removed?
If empty, then this will break. Types must have at least one field.

Reply:
It will cause a composition error, that is why I added this placeholder field here. We will add mutation fields to the public api soon, and we can then remove the placeholder fields.